pkg/auth/dependency/webapp/authenticate_provider_impl.go

```diff
@@ -607,6 +605,12 @@ func (p *AuthenticateProviderImpl) HandleSSOCallback(w http.ResponseWriter, r *h
 		})
 	}
 
+	oauthError := r.Form.Get("error")
+	if oauthError != "" {
+		err = sso.NewSSOFailed(sso.SSOUnauthorized, "login failed")
```

Comment:
I remember in the spec there is `error` and `error_description`. Maybe we should forward them to enable better diagnosis?

Reply:
`error` is an error code, therefore should not be displayed. I'll include `error_description` in the error message instead.

Reply:
I guess it is not harmful to show the error code as well.

Reply:
The UX can be confusing to end-user, since the error code often contains technical terms.

Reply:
OK. Let's revisit this later. I was thinking about using `<script>` to log the error code to console......